src/ripple/basics/impl/StringUtilities.cpp

```diff
@@ -89,11 +89,10 @@ uint64_t uintFromHex (std::string const& strSrc)
     return uValue;
 }
 
-// TODO Callers should be using beast::URL and beast::parse_URL instead.
 bool parseUrl (parsedURL& pUrl, std::string const& strUrl)
 {
     // scheme://username:password@hostname:port/rest
-    static boost::regex reUrl ("(?i)\\`\\s*([[:alpha:]][-+.[:alpha:][:digit:]]*)://([^/]+)(/.*)?\\s*?\\'");
+    static boost::regex reUrl ("(?i)\\`\\s*([[:alpha:]][-+.[:alpha:][:digit:]]*)://([^/]*)(/.*)?\\s*?\\'");
```

Comment:
I have a somewhat modified regular expression.

    static boost::regex reUrl(
        "(?i)\\`\\s*"
        "([[:alpha:]][-+.[:alpha:][:digit:]]*):" // scheme
        "//([^/]+)?"                             // hostname (possibly empty)
        "(/.*)?"                                 // rest
        "\\s*?\\'");

The difference between mine and yours is the hostname portion. I do: `([^/]+)?` and you do `([^/]*)`. I prefer mine because it explicitly matches the preceding expression exactly zero or one times, but I don't feel strongly about it.

Also, for another PR, I _this_ we should just improve this RE to properly crack the `user@host:domain:port` syntax instead of manually cracking it below.

Reply:
Functionally, I don't think there's a difference between the two hostname propositions.
`(X*)` greedily matches `X` zero or more times and stores what's found in the appropriate grouping. `(X+)?` greedily matches `X` one or more times zero or one times and stores what's found in the appropriate grouping. To me, the `(X*)` is easier to wrap my head around and explain, so if there's no objection, I'm going to stick with it.

I'm not so sure about breaking down the `host` syntax. I followed some of the code, and it's doing some complicated stuff there, including IPv4 vs IPv6 parsing, but I went ahead and made the ticket #2751.